securitybot/chat/slack.py

```diff
@@ -37,20 +41,33 @@ def _validate(self):
             raise ChatException('Unable to connect to Slack API.')
         logging.info('Connection to Slack API successful!')
 
-    def _api_call(self, method, **kwargs):
+    def _api_call(self, method, rate_limit_retry=True, **kwargs):
         # type: (str, **Any) -> Dict[str, Any]
         '''
         Performs a _validated_ Slack API call. After performing a normal API
-        call using SlackClient, validate that the call returned 'ok'. If not,
-        log and error.
+        call using SlackClient, validate that it has not been rate limited.  If it has,
+        retry every RATE_LIMIT_SLEEP until RATE_LIMIT_TRIES reached.
+        Validate that the call returned 'ok'. If not, log and error.
 
         Args:
             method (str): The API endpoint to call.
             **kwargs: Any arguments to pass on to the request.
         Returns:
             (dict): Parsed JSON from the response.
         '''
-        response = self._slack.api_call(method, **kwargs)
+        cur_try = 0
+        while True:
+            response = self._slack.api_call(method, **kwargs)
+            if cur_try > RATE_LIMIT_TRIES:
+                raise ChatException('Slack rate limit max tries reached.')
+
+            if response.get('error', '').lower() == 'ratelimited' and rate_limit_retry is True:
```

Comment:
```python
... and rate_limit_retry:  # ?
```

> Don’t compare boolean values to True or False using ==:
> 
> Yes:
> 
>     if greeting:
> 
> No:
> 
>     if greeting == True:
> 
> Worse:
> 
>     if greeting is True:

* https://pep8.org/#programming-recommendations

Reply:
Thank you.  I'm making the corrections.